pkg/skaffold/errors/errors.go

```diff
@@ -67,11 +64,12 @@ func ShowAIError(err error) error {
 	for _, v := range knownBuildProblems {
 		if v.regexp.MatchString(err.Error()) {
 			if suggestions := v.suggestion(skaffoldOpts); suggestions != nil {
-				return fmt.Errorf("%s. %s", v.description, concatSuggestions(suggestions))
+				return fmt.Errorf("%s. %s", strings.Trim(v.description(err), "."), concatSuggestions(suggestions))
```

Comment:
`strings.Trim(v.description(err), ".")`
can the `description` take care of the formatting and punctuation itself?

Reply:
the description is a func which returns a subset of error message from error. 
Moving it there, will force the developers to make sure, their regex trims "." which could be easily overlooked. 
Hence adding it here.

Reply:
I meant to say if you did https://github.com/GoogleContainerTools/skaffold/pull/4914#discussion_r505405657 description could stay a string.

Reply:
i responded above. I don't want it to be just a string. i wanted to extract information from the error since 
1.  error is wrapped in many places. we cannot use `errors.Unwrap` to solve this as we are relying on wrapped text ;) 
2.  still wont solve the issue where deploy context is minikube.